Vatinius iP

Vatinius opened the pull request:
# Evelyn Chatbot

> [!NOTE]
> Evelyn is still under development. Final product may vary.

> "With great power comes great responsibility." - Uncle Ben

## About
Evelyn is a command line application that allows you to keep track of you upcoming tasks in the form of:
- _Todos_
- _Deadlines_
- _Events_

## Getting started
To get started with Evelyn, all you need to do is:
1. Download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/117).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 🫨

And it is **FREE**

## Planned updates
- [x] Implement a find feature
- [ ] Implement a GUI for Evelyn

## Commands
Some basic commands are:
```
todo [task description]
deadline [task description] /by [date in YYYY-MM-DD] [Optional: time]
event [task description] /from [date in YYYY-MM-DD] [Optional: time] /to [date in YYYY-MM-DD] [Optional: time]
list (Lists out all the current tasks)
mark [index] (Marks the tasks at the specified index)
unmark [index] (Unmarks the tasks at the specified index)
delete [index] (deletes the tasks at the specified index)
find [keyword] (Unmarks the tasks at the specified index)
```

To get it started, find this method in the Evelyn class and run it!
``` Java
public static void main(String[] args) throws IOException {
        String dataFilePath = "src/main/data/evelyn.txt";
        new Evelyn(dataFilePath).run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Vatinius committed:
Added greeting and goodbye

Vatinius committed:
Added implemented feature

Vatinius committed:
Added variations for exit conditions

Vatinius committed:
Implemented Task class

Vatinius committed:
Updated logic in chatbot to handle mark and unmark

Vatinius committed:
Completed level 2

Vatinius committed:
Implemented Level 3

Vatinius committed:
Implemented todo functionality and class

Vatinius committed:
Implemented and added Deadline functionality and class

Vatinius committed:
Implemented deadline and event functionality and class

Vatinius committed:
Implemented Automated Text UI Testing

Vatinius committed:
Created exceptions to deal with invalid inputs and inputs that did not use the keywords properly

Vatinius committed:
Implemented delete feature

Vatinius committed:
Change Task to abstract class for implementation of write function

Vatinius committed:
Change formatting of Evelyn file to make it more readable

Vatinius committed:
Add save feature to chatbot to allow saving of tasks

Vatinius committed:
Edit import lines to proper convention

Vatinius committed:
Add test class to test branching

Vatinius committed:
Merge commit '973821f5b5a642a8f02f88b055486483bd21cb71' into branch-level-7

Vatinius committed:
Remove java test file

Vatinius committed:
Restore mark and unmark feature functionality

Vatinius committed:
Restore mark and unmark feature functionality

Vatinius committed:
Fix bug that saved data was not being input into list properly during startup of bot

Vatinius committed:
Patch save feature to enable saving of data

Vatinius committed:
Add date and time feature and fix bug whereby marked and unmarked tasks are not properly saved

Vatinius committed:
Add parser, storage, task list and ui classes with zero functionality

Vatinius committed:
Add Ui, Storage, Parser and TaskList classes to make programme more OOP

Vatinius committed:
Divide classes into packages

Vatinius committed:
Revert "Divide classes into packages"

This reverts commit 919274b70395e74012a028c08a7eb19d6042da43.

Vatinius committed:
Revert changes

Vatinius committed:
Fix bug whereby Evelyn class was not found

Vatinius committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-A-Gradle

Vatinius committed:
Add Gradle to project and fix bug whereby deleting a task does not decrement the number of tasks in the list

Vatinius committed:
Move Evelyn files into a package

Vatinius committed:
Evelyn: implement error handling for wrong date format

Evelyn can now present the user an error if they input a date that does not follow YYYY-MM-DD

JUnit: implement test cases for proper testing

Vatinius committed:
Implement test cases for all task types

Vatinius committed:
Package app as an executable JAR file so that it can be distributed easily

Vatinius committed:
Add JavaDoc comments to all the code.

Vatinius committed:
Edit code such that it complies with Java coding standard.

Vatinius committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/evelyn/command/Storage.java

Vatinius committed:
Edit Storage file to comply with coding standard.

Vatinius committed:
Implement a find feature that returns all tasks that contains the specified keyword.

Vatinius committed:
Remove .txt file that was used for testing.

Vatinius committed:
Update typo in README and UI files in the description of the delete command

Vatinius committed:
Implement checkstyle to detect coding style violations

Vatinius committed:
Set up runnable javafx application

Vatinius committed:
Implement gui to follow javafx tutorial part 3

ghos7ie commented on src/main/java/evelyn/command/Parser.java:
```diff
@@ -0,0 +1,76 @@
+ * Houses all the logic for parsing words that have been inputted
+ */
+public class Parser {
+    private TaskList lst;
```
```suggestion
    private TaskList list;
```

could update as such? according to code quality naming.

ghos7ie commented on src/main/java/evelyn/command/TaskList.java:
```diff
@@ -0,0 +1,216 @@
+            System.out.println(Ui.horizontalLine);
+            System.out.println("Got it. I've added this task:");
+            System.out.println("  " + task.toString());
+            System.out.println(this.listSize() > 1 ? "Now you have " + this.listSize() + " tasks in this list"
```
Could shift printing number of tasks into a method of its own, allows for easier maintenance!

ghos7ie commented on src/main/java/evelyn/command/Ui.java:
```diff
@@ -0,0 +1,102 @@
+    public void start() {
+        this.isChatting = true;
+        System.out.println(horizontalLine);
+        System.out.println("Hi! I am evelyn.Evelyn");
```
might be easier to maintain your code by compiling all the strings into a StringBuilder obj, then printing that instead

ghos7ie reviewed (commented):
Code looks very clean and well arranged! Can't find any other issues.

ghos7ie commented on src/main/java/evelyn/command/TaskList.java:
```diff
@@ -0,0 +1,216 @@
+     * @param data String representation of a task form the save data.
+     * @throws IOException Exception thrown if the file data input is erroneous.
+     */
+    private void fileDataToList(String data) throws IOException {
```
```suggestion
    private void writeFileDataToList(String data) throws IOException {
```
I think this might make the purpose of the method more obvious and also in line with the code quality standards mentioned!

DanzaSeah commented on src/main/java/evelyn/command/Ui.java:
```diff
@@ -0,0 +1,102 @@
+
+    /**
+     * Constructor of a Ui object.
+     * @param parser Parser used for the chatbot.
```
I noticed that there’s no blank line between the description and the @param tag in the Javadoc for the constructor, and in other classes as well. Maybe adding a line between would be helpful.

DanzaSeah commented on src/main/java/evelyn/command/TaskList.java:
```diff
@@ -0,0 +1,216 @@
+     * @param keyword
+     */
+    public void find(String keyword) {
+        boolean foundSomething = false;
```
Is there an additional space indentation at the beginning of the statement?

DanzaSeah commented on src/main/java/evelyn/task/Task.java:
```diff
@@ -0,0 +1,48 @@
+     * @param desctiption Description of a Task.
+     * @param isMarked Status of a Task.
+     */
+    public Task (String desctiption, boolean isMarked) {
```
Perhaps there has been a mistype here and the following line, where you may have entered "desctiption" instead of "description".
```suggestion
    public Task (String description, boolean isMarked) {
```

DanzaSeah commented on src/main/java/evelyn/task/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+     * @param deadline Deadline of a Deadline task.
+     * @param isMarked Status of a Deadline task.
+     */
+    public Deadline(String desccription, String deadline, boolean isMarked) {
```
Perhaps there has been a mistype here and the following line, where you may have entered "desccription" instead of "description".
```suggestion
    public Deadline(String description, String deadline, boolean isMarked) {
```

DanzaSeah commented on src/main/java/evelyn/task/Task.java:
```diff
@@ -0,0 +1,48 @@
+        this.isMarked = true;
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("    " + this.toString());
+    };
```
I noticed there was a couple of places where there was an additional semicolon after the closing curly braces of a method definition, maybe it would be better to remove it?

DanzaSeah reviewed (commented):
Great work on this project! The overall structure is well thought-out, and it's clear that a lot of effort went into building a solid foundation for the Evelyn chatbot. The code is mostly easy to follow, and I appreciate the use of comments to describe the purpose of different sections, which helps with readability. Other than some nits to work out, it's a great job!

Vatinius committed:
Implement working gui fully for Evelyn

Vatinius committed:
Fix all style errors

Vatinius committed:
Add styling to evelyn

Vatinius committed:
Add assertions to evelyn

There were previously no assertions used in evelyn.
Adding assertions increases reliability of evelyn to detect runtime errors.

Vatinius committed:
Refactor tasklist file

There were redundant castings in tasklist.
I removed them.

Vatinius committed:
Refactor parser code

Vatinius committed:
Merge pull request #1 from Vatinius/branch-A-CodeQuality

Refactor parser code

Vatinius committed:
Implement friendlier syntax

Vatinius committed:
Organise code to conform to s.l.a.p

Vatinius committed:
Refactor TaskList file to improve SLAP

Vatinius committed:
Comment out text based UI file

Vatinius committed:
Edit header comments

Vatinius committed:
Implement README and fix bug

Implement a README for Evelyn.

Fix a bug for Todo tasks whereby
tasks that are more than
one word in length were
unable to be input.

Vatinius committed:
Update README.md

Vatinius committed:
Update README.md

Vatinius committed:
Improve JUnit test cases

Vatinius committed:
Fix gradle file

Vatinius committed:
Fix relative file path

Vatinius committed:
Fix keyword issue

Vatinius committed:
Improve syntax feature

Vatinius committed:
Change user and Evelyn images

Vatinius committed:
Fix mark task bug

Vatinius committed:
Fix typo in intro text

Vatinius committed:
Update README file

Vatinius committed:
Update readme file

Vatinius committed:
Update readme file

Vatinius committed:
Update testcases

Vatinius committed:
Remove unused code

Vatinius committed:
Fix typo mistake

Vatinius committed:
Fix more typos

Vatinius committed:
Fix delete bug

Vatinius committed:
Fix typo in error

Vatinius committed:
Update user guide
